Changes to interceptor engagement

cjbehm opened the pull request:
I don't know if you're taking PRs but I thought I'd open one with the changes I made. Solves the weird acrobatics I mentioned in the forum thread and avoids the all-knowing engageGroup task type.

Reduces the knowledge of the intercepting group
- Only revector if the target has a large detected change in position
- Do not use engageGroup task since it knows where the group is
- Remove intruders from the list if they are no longer intruding
- Allow interceptors to give up

cjbehm committed:
Changes to intercept engagement

Reduces the knowledge of the intercepting group
- Only revector if the target has a large detected change in position
- Do not use engageGroup task since it knows where the group is
- Remove intruders from the list if they are no longer intruding
- Allow interceptors to give up

lukrop commented:
Of course I do. Thanks! I'll merge this ASAP.

lukrop commented on GCICAP.lua:
```diff
@@ -381,6 +384,7 @@ do
       local side = gcicap.coalitionToSide(group:getCoalition())
       local f = {}
       f.side = side
+      f.giveUp = false
```
please use give_up instead of giveUp

lukrop commented on GCICAP.lua:
```diff
@@ -550,20 +557,34 @@ do
         end
 
-        gcicap.log:info("Vectoring $1 to $2 ($3)", self.group:getName(),
-                 intruder.group:getName(), target:getName())
```
Is there a reason why you removed the log message?

cjbehm committed:
Address comments re: giveUp vs give_up

cjbehm committed:
Make sure to allow re-tasking by GCI controller

Need to reset the give_up flag if a flight (like a GCI flight) is
re-tasked.

cjbehm commented on GCICAP.lua:
```diff
@@ -522,6 +528,7 @@ do
                           command = "local group = ...\
                           local flight = gcicap.Flight.getFlight(group)\
                           if flight then\
+                            flight.give_up = true\
```
Found a bug with this and GCI flights. This triggers really quickly and I'm not sure why yet. Possibly because the flights start on the ground - maybe they need two points at the target, a move and then the engage. I'd hold off merging until I address it, it makes GCI flights almost completely broken.

cjbehm committed:
Fix GCI spawning

If there were no flights available to task for an intercept, GCI
actually would not be fired at all. Now try to spawn GCI flights when
any intruder needs to be intercepted.

cjbehm committed:
Remove intruders outside of loop

Don't modify the intruder list while looping through the intruder list.
Record and then remove after the loop.

cjbehm committed:
Fix GCI take-off/immediate land

The change from EnageGroup to EngageTargets had an unintended
consequence for GCI flights. They would take off an immediately land.

Added a CAP task at the point of intercept. This seems to give expected
behavior.

cjbehm commented:
I found a bug with GCI spawning while I was working on this, so I fixed that too. I considered doing it in a separate PR, so if you want me to make this PR smaller by doing so I can figure out how to go backwards ;)

lukrop commented:
If you would create a separate PR for it I could merge it sooner since the bug you found in your new code isn't fixed yet I assume?

cjbehm commented:
Sorry, wasn't clear on what fixes were in the most recent commits I made. 08eac30, 948781b fix the bug I created and commit 237da8b fixes a bug in the original code that wouldn't spawn GCI flights if CAP flights were off.

I think we're clear to merge - I haven't seen any obvious bugs in my test mission.
